src/maxdiffusion/generate_sdxl.py

```diff
@@ -68,15 +68,20 @@ def loop_body(step, args, model, pipeline, added_cond_kwargs, prompt_embeds, gui
     added_cond_kwargs=added_cond_kwargs
   ).sample
 
-  def apply_classifier_free_guidance(noise_pred, guidance_scale, guidance_rescale):
+  def apply_classifier_free_guidance(noise_pred, guidance_scale):
     noise_pred_uncond, noise_prediction_text = jnp.split(noise_pred, 2, axis=0)
     noise_pred = noise_pred_uncond + guidance_scale * (noise_prediction_text - noise_pred_uncond)
-    if guidance_rescale > 0:
-      noise_pred = rescale_noise_cfg(noise_pred, noise_prediction_text, guidance_rescale)
-    return noise_pred
+    return noise_pred, noise_prediction_text
 
   if config.do_classifier_free_guidance:
-    noise_pred = apply_classifier_free_guidance(noise_pred, guidance_scale, guidance_rescale)
+    noise_pred, noise_prediction_text = apply_classifier_free_guidance(noise_pred, guidance_scale)
+
+  noise_pred = jax.lax.cond(
+      guidance_rescale[0] > 0,
+      lambda _: rescale_noise_cfg(noise_pred, noise_prediction_text, guidance_rescale),
+      lambda _: noise_pred,
+      operand=None
+  )
```

Comment:
nit: Since this call is being used in both of the files is there a way to put it in a common utils file ?

Reply:
I don't think the jax conditional needs to be refactored. The main code is in the rescale_noise_cfg fn.